contour_example_serve.py

```diff
@@ -0,0 +1,29 @@
+from bokeh.palettes import RdYlGn
+from bokeh.plotting import figure, curdoc
+from bokeh.plotting.contour import contour_data_dicts
+import numpy as np
+
+def calc_z(timestep):
+    return np.sin(x - 0.1*timestep)*np.cos(y - 0.02*timestep)
+
+x, y = np.meshgrid(np.linspace(0, 10, 20), np.linspace(0, 6, 10))
+z = calc_z(0)
+nlevels = 10
+levels = np.linspace(-1.0, 1.0, nlevels)
+fill_color = RdYlGn[nlevels-1]
+line_color = "black"
+timestep = 0
+
+def callback():
+    global timestep
+    timestep += 1
```

Comment:
Just FYI see the relatively under-used functions in `bokeh.driving` to avoid explict global management like this, e.g. https://docs.bokeh.org/en/latest/docs/reference/driving.html#bokeh.driving.count

Reply:
Nice!

Also, I bet there are some really useful internal functions somewhere to help deal with scalar vs vector properties, I just haven't found them yet.